File: 2.link-list/13.create-linked-list/create-linked-list.js
// Write a function, createLinkedList, that takes in an array of values as an argument. The function should create a linked list containing each element of the array as the values of the nodes. The function should return the head of the linked list.

class Node {
    constructor(val) {
        this.val = val;
        this.next = null;
    }
}

const createLinkedList = values => {
    let head = new Node(null);
    let cur = head;
    for (let i = 0; i < values.length; i++) {
        cur.next = new Node(values[i]);
        cur = cur.next;
    }
    return head.next;
};

console.log(createLinkedList(['H', 'E', 'Y']));
// h -> e -> y

File: 2.link-list/13.create-linked-list/create-linked-list.go
// Write a function, createLinkedList, that takes in an array of values as an argument. The function should create a linked list containing each element of the array as the values of the nodes. The function should return the head of the linked list.

package main

import "fmt"

type Node struct {
	value interface{}
	next  *Node
}

func NewNode(value interface{}) *Node {
	return &Node{value: value}
}

func printLinkedList(head *Node) {
	if head == nil {
		return
	}
	fmt.Println(head.value)
	printLinkedList(head.next)
}

func createLinkedList(arr []interface{}) *Node {
	head := NewNode(nil)
	cur := head
	for _, v := range arr {
		cur.next = NewNode(v)
		cur = cur.next
	}
	return head.next
}

func main() {
	printLinkedList(createLinkedList([]interface{}{}))
	// h -> e -> y
}
